rcl/test/rcl/test_subscription.cpp

```diff
@@ -528,6 +595,139 @@ TEST_F(CLASSNAME(TestSubscriptionFixture, RMW_IMPLEMENTATION), test_subscription
   }
 }
 
+/* Test for all failure modes in subscription take with loaned messages function.
+ */
+TEST_F(CLASSNAME(TestSubscriptionFixture, RMW_IMPLEMENTATION), test_bad_take_loaned_message) {
+  const char * topic = "rcl_loan";
+  const rosidl_message_type_support_t * ts = ROSIDL_GET_MSG_TYPE_SUPPORT(test_msgs, msg, Strings);
+  rcl_subscription_options_t subscription_options = rcl_subscription_get_default_options();
+
+  rcl_subscription_t subscription = rcl_get_zero_initialized_subscription();
+  rmw_ret_t ret = rcl_subscription_init(
+    &subscription, this->node_ptr, ts, topic,
+    &subscription_options);
+  ASSERT_EQ(RMW_RET_OK, ret) << rcl_get_error_string().str;
+
+  test_msgs__msg__Strings * loaned_message = nullptr;
+  void ** type_erased_loaned_message_pointer = reinterpret_cast<void **>(&loaned_message);
+  EXPECT_EQ(
+    RCL_RET_SUBSCRIPTION_INVALID, rcl_take_loaned_message(
+      nullptr, type_erased_loaned_message_pointer, nullptr, nullptr));
```

Comment:
Here and below for invalid parameters tests using `nullptr` it would be great if you can follow the idea that @brawner gave me here: https://github.com/ros2/rcl/pull/732#discussion_r467264915